core/src/main/java/org/jruby/ext/ripper/RipperLexer.java

```diff
@@ -2349,7 +2350,7 @@ public boolean tokenAddMBC(int first_byte, ByteList buffer) {
         int length = precise_mbclen();
 
         if (length <= 0) {
-            compile_error("invalid multibyte char (" + getEncoding().getName() + ")");
+            compile_error("invalid multibyte char (" + Arrays.toString(getEncoding().getName()) + ")");
```

Comment:
I didn't notice this before, but this should just be `getEncoding()`. The default toString for Encoding is to return the name of the encoding.